### PR TITLE
[CELEBORN-1615] Start the http server after all handlers added

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

Start the http server after all handlers added.

### Why are the changes needed?
Do not expose the RESTful server before all handlers initialized.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Existing UT.

### CODE REVIEW

**Review by FMX — APPROVED** [review]
LGTM. I think this change has no side effects.

**Review by cxzl25 — APPROVED** [review]

**Comment by FMX** [comment]
After some digging,  this PR might not find the root cause.  The master class is a subclass of the Service class.
The master object creates a master instance and calls its initialize method.
截屏2024-09-27 16 04 03
Master instance calls its superclass method of initialize.
截屏2024-09-27 16 04 44
HttpService calls its superclass method of initialize.
截屏2024-09-27 16 05 27

Those initialization methods are invoked in the same thread.

We should find out the root cause and fix this problem.

**Review by FMX — CHANGES_REQUESTED** [review]
Maybe find out the root cause for this problem will be better.

**Comment by turboFei** [comment]
Hi @waitinfuture 

I have changed the PR description and title.

Just start to expose the RESTful server after all handlers initialization finished.

File: service/src/main/scala/org/apache/celeborn/server/common/HttpService.scala [hunk 1]
```diff
@@ -348,9 +347,13 @@ abstract class HttpService extends Service with Logging {
 
     if (metricsSystem.running) {
       metricsSystem.getServletContextHandlers.foreach { handler =>
+        logInfo(s"Adding metrics servlet handler with path ${handler.getContextPath}")
         httpServer.addHandler(handler)
       }
     }
+
+    // start the http server after all handlers are added
+    httpServer.start()
```
[the remark this answers is not in the document]
Comment (RexXiong) [reply service/src/main/scala/org/apache/celeborn/server/common/HttpService.scala:356]: > how about starting/exposing the http endpoint after all handlers added? @waitinfuture
> 
> In our use case, there is an agent to collect the celeborn metrics periodically.

Sounds reasonable

**Review by RexXiong — APPROVED** [review]
LGTM

**Pull request closed without merging**